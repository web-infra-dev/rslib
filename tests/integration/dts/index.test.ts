import { join } from 'node:path';
import { buildAndGetResults } from 'test-helper';
import { describe, expect, test } from 'vitest';

describe('dts when bundle: false', () => {
  test('basic', async () => {
    const fixturePath = join(__dirname, 'bundle-false', 'basic');
    const { files, contents } = await buildAndGetResults(fixturePath, 'dts');

    expect(files.esm).toMatchInlineSnapshot(`
      [
        "<ROOT>/tests/integration/dts/bundle-false/basic/dist/esm/index.d.ts",
        "<ROOT>/tests/integration/dts/bundle-false/basic/dist/esm/sum.d.ts",
        "<ROOT>/tests/integration/dts/bundle-false/basic/dist/esm/utils/numbers.d.ts",
        "<ROOT>/tests/integration/dts/bundle-false/basic/dist/esm/utils/strings.d.ts",
      ]
    `);

    expect(files.cjs).toMatchInlineSnapshot(`
      [
        "<ROOT>/tests/integration/dts/bundle-false/basic/dist/cjs/index.d.ts",
        "<ROOT>/tests/integration/dts/bundle-false/basic/dist/cjs/sum.d.ts",
        "<ROOT>/tests/integration/dts/bundle-false/basic/dist/cjs/utils/numbers.d.ts",
        "<ROOT>/tests/integration/dts/bundle-false/basic/dist/cjs/utils/strings.d.ts",
      ]
    `);

    expect(contents.esm).toMatchSnapshot();
  });

  test('dts false', async () => {
    const fixturePath = join(__dirname, 'bundle-false', 'false');
    const { files } = await buildAndGetResults(fixturePath, 'dts');

    expect(files.esm).toMatchInlineSnapshot('undefined');
  });

  test('distPath', async () => {
    const fixturePath = join(__dirname, 'bundle-false', 'dist-path');
    const { files } = await buildAndGetResults(fixturePath, 'dts');

    expect(files.esm).toMatchInlineSnapshot(`
      [
        "<ROOT>/tests/integration/dts/bundle-false/dist-path/dist/custom/index.d.ts",
        "<ROOT>/tests/integration/dts/bundle-false/dist-path/dist/custom/sum.d.ts",
        "<ROOT>/tests/integration/dts/bundle-false/dist-path/dist/custom/utils/numbers.d.ts",
        "<ROOT>/tests/integration/dts/bundle-false/dist-path/dist/custom/utils/strings.d.ts",
      ]
    `);
  });

  test('abortOnError: false', async () => {
    const fixturePath = join(__dirname, 'bundle-false', 'abort-on-error');
    const { isSuccess } = await buildAndGetResults(fixturePath, 'dts');

    expect(isSuccess).toBe(true);
  });

  test('autoExtension: true', async () => {
    const fixturePath = join(__dirname, 'bundle-false', 'auto-extension');
    const { files } = await buildAndGetResults(fixturePath, 'dts');

    expect(files.cjs).toMatchInlineSnapshot(`
      [
        "<ROOT>/tests/integration/dts/bundle-false/auto-extension/dist/cjs/index.d.cts",
        "<ROOT>/tests/integration/dts/bundle-false/auto-extension/dist/cjs/sum.d.cts",
        "<ROOT>/tests/integration/dts/bundle-false/auto-extension/dist/cjs/utils/numbers.d.cts",
        "<ROOT>/tests/integration/dts/bundle-false/auto-extension/dist/cjs/utils/strings.d.cts",
      ]
    `);
  });
});

describe('dts when bundle: true', () => {
  test('basic', async () => {
    const fixturePath = join(__dirname, 'bundle', 'basic');
    const { entryFiles, entries } = await buildAndGetResults(
      fixturePath,
      'dts',
    );

    expect(entryFiles.esm).toMatchInlineSnapshot(
      `"<ROOT>/tests/integration/dts/bundle/basic/dist/esm/index.d.ts"`,
    );

    expect(entryFiles.cjs).toMatchInlineSnapshot(
      `"<ROOT>/tests/integration/dts/bundle/basic/dist/cjs/index.d.ts"`,
    );

    expect(entries).toMatchSnapshot();
  });

  test('dts false', async () => {
    const fixturePath = join(__dirname, 'bundle', 'false');
    const { entryFiles } = await buildAndGetResults(fixturePath, 'dts');

    expect(entryFiles.esm).toMatchInlineSnapshot('undefined');
  });

  test('distPath', async () => {
    const fixturePath = join(__dirname, 'bundle', 'dist-path');
    const { entryFiles } = await buildAndGetResults(fixturePath, 'dts');

    expect(entryFiles.esm).toMatchInlineSnapshot(
      `"<ROOT>/tests/integration/dts/bundle/dist-path/dist/custom/index.d.ts"`,
    );
  });

  test('abortOnError: false', async () => {
    const fixturePath = join(__dirname, 'bundle', 'abort-on-error');
    const { isSuccess } = await buildAndGetResults(fixturePath, 'dts');

    expect(isSuccess).toBe(true);
  });

  test('autoExtension: true', async () => {
    const fixturePath = join(__dirname, 'bundle', 'auto-extension');
    const { entryFiles } = await buildAndGetResults(fixturePath, 'dts');

    expect(entryFiles.cjs).toMatchInlineSnapshot(
      `"<ROOT>/tests/integration/dts/bundle/auto-extension/dist/cjs/index.d.cts"`,
    );
  });

  test('bundleName -- set source.entry', async () => {
    const fixturePath = join(__dirname, 'bundle', 'bundle-name');
    const { entryFiles } = await buildAndGetResults(fixturePath, 'dts');

    expect(entryFiles.esm).toMatchInlineSnapshot(
      `"<ROOT>/tests/integration/dts/bundle/bundle-name/dist/esm/bundleName.d.ts"`,
    );
  });

  test('entry is an absolute path', async () => {
    const fixturePath = join(__dirname, 'bundle', 'absolute-entry');
    const { entryFiles } = await buildAndGetResults(fixturePath, 'dts');

    expect(entryFiles.esm).toMatchInlineSnapshot(
      `"<ROOT>/tests/integration/dts/bundle/absolute-entry/dist/esm/index.d.ts"`,
    );
  });
});
