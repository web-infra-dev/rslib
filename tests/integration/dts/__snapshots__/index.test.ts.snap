// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`dts when bundle: false > basic 3`] = `
{
  "<ROOT>/tests/integration/dts/bundle-false/basic/dist/esm/index.d.ts": "export * from './utils/numbers';
export * from './utils/strings';
export * from './sum';
",
  "<ROOT>/tests/integration/dts/bundle-false/basic/dist/esm/sum.d.ts": "export declare const numSum: number;
export declare const strSum: string;
",
  "<ROOT>/tests/integration/dts/bundle-false/basic/dist/esm/utils/numbers.d.ts": "export declare const num1 = 1;
export declare const num2 = 2;
export declare const num3 = 3;
",
  "<ROOT>/tests/integration/dts/bundle-false/basic/dist/esm/utils/strings.d.ts": "export declare const str1 = "str1";
export declare const str2 = "str2";
export declare const str3 = "str3";
",
}
`;

exports[`dts when bundle: true > basic 3`] = `
{
  "cjs": "export declare const num1 = 1;

export declare const num2 = 2;

export declare const num3 = 3;

export declare const numSum: number;

export declare const str1 = "str1";

export declare const str2 = "str2";

export declare const str3 = "str3";

export declare const strSum: string;

export { }
",
  "esm": "export declare const num1 = 1;

export declare const num2 = 2;

export declare const num3 = 3;

export declare const numSum: number;

export declare const str1 = "str1";

export declare const str2 = "str2";

export declare const str3 = "str3";

export declare const strSum: string;

export { }
",
}
`;
